# Ejercicios de práctica clase 2

Determinando tipos de datos

In [10]:
"""
Revisando cómo se comportan los datos y las buenas prácticas
Viendo diferencias entre el comportamiento de los datos
"""

# Esto es un comentario :v
print("Hola")

Hola


In [6]:
print("mundo")

mundo


In [22]:
print("Hola mundo")

Hola mundo


In [12]:
print("Hola", "mundo")

Hola mundo


In [21]:
print("Hola", end=" ")

Hola 

In [13]:
print("Hola", "jovenes", "como", "amanecen")

Hola jovenes como amanecen


In [14]:
print("Hola jovenes como amanecen")

Hola jovenes como amanecen


In [15]:
print("Hola", "jovenes", "como", "amanecen", sep=",")

Hola,jovenes,como,amanecen


In [16]:
print("Hola", "jovenes", "\ncomo", "amanecen", sep=",")

Hola,jovenes,
como,amanecen


In [23]:
print("Hola" + "Maria")

HolaMaria


In [24]:
print("Hola" + " " + "Maria")

Hola Maria


In [17]:
a = 7+5

In [18]:
print(a)

12


In [19]:
print(7+5)

12


In [20]:
print(f"7+5 = {a}")

7+5 = 12


In [25]:
print(type(a))

<class 'int'>


# Entrada y Salida de Datos

In [4]:
nombre = input("Teclee su nombre")

Teclee su nombre Alice


In [5]:
apellido = input("Teclee su apellido")

Teclee su apellido The 9th


In [6]:
apellido = input(f"ahora {nombre} teclee su apellido")

ahora Alice teclee su apellido The 9th


In [8]:
print("Nombre",nombre, end = " ")

Nombre Alice 

In [9]:
edad = int(input(f"Muy bien {nombre} {apellido}, ahora dime tu edad"))

Muy bien Alice The 9th, ahora dime tu edad 102


In [10]:
qq = edad + 2

In [14]:
print(qq)

104


### Si no se determina el tipo de dato:

In [23]:
a = input("Entre un numero entero")

Entre un numero entero 5


In [24]:
b = input("Entre otro numero entero")

Entre otro numero entero 5


In [25]:
print(a+b)

55


En general, todo dato que se lea con la instrucción input el programa Python lo almacena como una hilera de caracteres (tipo string). 

### Determinando el tipo de dato

In [20]:
a = int(input("Entre un numero entero"))

Entre un numero entero 5


In [21]:
b = int(input("Entre otro numero entero"))

Entre otro numero entero 5


In [22]:
print(a+b)

10


### Qué es una variable

In [30]:
import math

In [31]:
math.pi

3.141592653589793

In [32]:
True + False == 0

False

In [39]:
b = ["string", 12, 0.01, [ "other_list" ]]

In [4]:
if (True or False):
    print("Hello, I’m ", 21, " years old")

Hello, I’m  21  years old


In [41]:
if True or False:
    print("Hello, I’m " + 21 +  " years old")

TypeError: can only concatenate str (not "int") to str

In [44]:
for i in range(2,30,2):
    print(i)

2
4
6
8
10
12
14
16
18
20
22
24
26
28


# Variables

## Reglas para declarar una variable

Debido a la manera en la que funciona el código de programación, se deben respetar ciertas reglas a la hora de declarar variables.

i.    El identificador no puede empezar por dígitos ni caracteres especiales [7] (a excepción del guion bajo “_”).

ii.    El resto del identificador no debe contener espacios ni caracteres especiales (a excepción del guion bajo “_”), pero sí puede contener dígitos.

iii.    Los identificadores son sensibles a las mayúsculas, es decir, la variable myvariable se identificará como diferente a myVariable. Nótese que la diferencia de las variables es la v en mayúscula.

##  Convenciones para declarar una variable

Un grupo de personas puede establecer un acuerdo para que todos hagan algo de la misma forma, ya sea por eficiencia, facilidad o incluso comodidad. A esto se lo llama convención. [8] Para declarar una variable, por ejemplo, se han establecido varias convenciones con el fin de mejorar su lectura y distinción.



i.    No utilizar una simple letra para una variable.


ii.    En Python, la más famosa es Snake Case, que consiste en que todo va en minúscula y las palabras están separadas por guiones bajos.

my_variable = 18


iii.    Si el objetivo de una variable es funcionar como constante, o sea que su valor nunca cambiará, se escribe todo en mayúscula y las palabras se separan por guiones bajos.

PI_NUMBER = 3.1416

# Tipos de datos

## Preguntas de interés

1. ¿Hay alguna manera de convertir un dato numérico decimal a entero?: sí, con el método round():

The round() function returns a floating point number that is a rounded version of the specified number, with the specified number of decimals.

The default number of decimals is 0, meaning that the function will return the nearest integer.


2. ¿Hay diferencias entre los datos textuales encerrados en diferentes tipos y cantidades de comillas?

Las cadenas de texto se pueden delimitar con comillas simples (') o con comillas dobles ("), funcionan de la misma manera y no hay diferencia en su resultado.
Lo único es que las cadenas se deben cerrar con las mismas comillas con las que se abrieron, de lo contrario estaremos cometiendo un error de sintaxis.


3. ¿Cómo se ponen comillas dentro de un tipo de dato textual?
Se debe encerrar la palabra/palabras/mensaje/texto a utilizar, ej: "Esto es un texto utilizando comillas"


4. ¿Qué sucede si se asigna una variable a otra variable?
Si se tiene por ejemplo:
a = 2
b = a
Lo que está pasando es que b se va a referir a a y al espacio de memoria donde está almacenado esta

5. Pruebe type() con diferentes datos.

In [5]:
type(3)

int

In [6]:
type(2.5)

float

In [8]:
type(False)

bool

In [9]:
type("Esto es un texto")

str

In [10]:
print(type(2))

<class 'int'>
